pkg/compact/compact.go

```diff
@@ -470,6 +474,87 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	PlanProgressCalc(ctx context.Context) error
+}
+
+type DefaultPlanSim struct {
+	grouper               Grouper
+	planner               Planner
+	sy                    *Syncer
+	numberOfIterations    prometheus.Gauge
```

Comment:
Let's not use global level metrics. We should use group level metrics so that we can get number of iterations to do for each group and also sum of total iterations of all groups